### ISSUE TITLE
Trying to set exported parameter of inherited scene fails

### ISSUE THREAD

**Issue by blackears** [godotengine/godot#78259]
### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 10, Compatability

### Issue description

Encountering a strange issue where I'm setting a parameter of a class and it just returns null whenever I try to access it.  This parameter is an exported parameter of a class that is the class that is the root class of an inherited scene.  I'm not sure if that is related to this issue or not.  Anyway, I've tried setting this parameter in the editor and while the editor lets me set it, when I try to read it later on it returns null.  Even if I assign these values in the _ready() method, they are still null when I try to read them later.

### Steps to reproduce

Open the attached project in the editor.  The class having the problem is res://scene/column_down_stack.tscn.  It has a property called `@export var up_stack:NodePath`.  There are several instances of this on the scene that provides the card table: res://scene/card_table.tscn.  Notice that all the col_down_* classes have their Up Stack value set.  I've also made doubly sure by setting this explicitly in the _ready method.  However, whenever `ColumnDownStack._on_selected()` is called (this happens if the user clicks on a face down pile in the columns area)), the `up_stack` parameter is null.  There is nothing else that sets this parameter, so I don't know why this is happening.  My only guess is that it has something to do with Godot getting mixed up due to the scene being inherited from the more general card stack scene.

### Minimal reproduction project

[godotSolitaire.zip](https://github.com/godotengine/godot/files/11754405/godotSolitaire.zip)

**Comment by blackears** [comment]
That worked.  I guess it was my fault after all.  Still, it would be helpful if the editor provided a warning when the user does this.

**Comment by dalexeev** [comment]
> Still, it would be helpful if the editor provided a warning when the user does this.

There is a warning, but it's not very noticeable and for the compiler it is a local variable:
![](https://github.com/godotengine/godot/assets/47700418/350a591b-ddce-42cb-a501-5d9c802eed0c)

After #73427, there will be an error:
![](https://github.com/godotengine/godot/assets/47700418/d584a9cf-55e5-4f1c-ab2b-0a2e40370fc7)

**Comment by dalexeev** [comment]
Looks like the issue is resolved, so I'm closing this.

* The bug with incorrect/non-obvious variable shadowing is tracked by #54944.
* The proper error message will be implemented by #73427.
* Or alternatively, if we reject the PR, this code will work in an intuitive way after another PR.
